### ISSUE TITLE
Two nearly identical lines were formatted surprisingly differently

### ISSUE THREAD

**Issue by sgrif** [rust-lang-nursery/rustfmt#1715]
Checking Diesel's codebase against the new defaults, I noticed this in the diff:

```patch
-pub type Between<Lhs, Rhs> = super::operators::Between<Lhs,
-    super::operators::And<AsExpr<Rhs, Lhs>, AsExpr<Rhs, Lhs>>>;
-pub type NotBetween<Lhs, Rhs> = super::operators::NotBetween<Lhs,
-    super::operators::And<AsExpr<Rhs, Lhs>, AsExpr<Rhs, Lhs>>>;
+pub type Between<Lhs, Rhs> = super::operators::Between<
+    Lhs,
+    super::operators::And<
+        AsExpr<Rhs, Lhs>,
+        AsExpr<Rhs, Lhs>,
+    >,
+>;
+pub type NotBetween<Lhs, Rhs> = super::operators::NotBetween<
+    Lhs,
+    super::operators::And<
+        AsExpr<
+            Rhs,
+            Lhs,
+        >,
+        AsExpr<
+            Rhs,
+            Lhs,
+        >,
+    >,
+>;
```

It's rather surprising that `AsExpr<Rhs, Lhs>` is broken up onto multiple lines in the second case, but not the first.

**Comment by topecongiro** [comment]
This is a duplicate of #1683, and should be fixed via #1717. Closing since this is a duplicate, although thank you for raising an issue!

**Comment by sgrif** [comment]
Oh lol literally the exact same example. Sorry I didn't see the other issue!
